flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -16,15 +16,202 @@
 
 package com.google.cloud.flink.bigquery.sink.serializer;
 
+import com.google.api.client.util.Preconditions;
 import com.google.cloud.flink.bigquery.sink.exceptions.BigQuerySerializationException;
 import com.google.protobuf.ByteString;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.DynamicMessage;
+import org.apache.avro.LogicalType;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericRecord;
 
+import javax.annotation.Nullable;
+
+import java.nio.ByteBuffer;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 /** Serializer for converting Avro's {@link GenericRecord} to BigQuery proto. */
 public class AvroToProtoSerializer implements BigQueryProtoSerializer<GenericRecord> {
 
+    private Descriptor descriptor;
+    private static final Map<Schema.Type, UnaryOperator<Object>> PRIMITIVE_TYPE_ENCODERS;
+
+    /*
+     * Map Avro Schema Type to an Encoding function which converts AvroSchema Primitive
+     * Type to Dynamic Message.
+     *
+     * PRIMITIVE_TYPE_ENCODERS: A Map containing mapping from Primitive Avro Schema Type with encoder function.
+     */
+    static {
+        PRIMITIVE_TYPE_ENCODERS = new EnumMap<>(Schema.Type.class);
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.INT, UnaryOperator.identity()); // INT -> INT
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.LONG, UnaryOperator.identity());
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.DOUBLE, UnaryOperator.identity());
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.BOOLEAN, UnaryOperator.identity());
+        PRIMITIVE_TYPE_ENCODERS.put(
+                Schema.Type.FLOAT,
+                o -> Float.parseFloat(String.valueOf((float) o))); // FLOAT -> FLOAT
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.STRING, Object::toString);
+        PRIMITIVE_TYPE_ENCODERS.put(Schema.Type.ENUM, Object::toString);
+        PRIMITIVE_TYPE_ENCODERS.put(
+                Schema.Type.FIXED, o -> ByteString.copyFrom(((GenericData.Fixed) o).bytes()));
+        PRIMITIVE_TYPE_ENCODERS.put(
+                Schema.Type.BYTES, o -> ByteString.copyFrom(((ByteBuffer) o).array()));
+    }
+
+    /**
+     * Initializer for the Serializer.
+     *
+     * @param bigQuerySchemaProvider BigQuerySchemaProvider for the Sink Table ({@link
+     *     BigQuerySchemaProvider} object)
+     */
+    public void init(BigQuerySchemaProvider bigQuerySchemaProvider) {
+        Preconditions.checkNotNull(
+                bigQuerySchemaProvider,
+                "BigQuerySchemaProvider not initialized before initializing Serializer.");
```

Comment:
Since this is a null check, a better error message would be: `BigQuerySchemaProvider not found while initializing Serializer`

Reply:
Fixed.